### ISSUE TITLE
disable rules for certain file paths or extensions

### ISSUE THREAD

**Issue by alexkrolick** [jest-community/eslint-plugin-jest#242]
I'm using Cypress and Jest with the ESLint plugins for both, and the Jest plugin is complaining about `consistent-test-it` and `expect-expect` inside the .spec.js files which are Cypress tests (Mocha syntax).

Being able to specify the file extension to lint would allow these rules to live together in one config.

**Comment by macklinu** [comment]
I am curious about this too, having rule into a similar issue in a personal project, but I have not yet explored using [ESLint's `overrides` option](https://eslint.org/docs/user-guide/configuring#disabling-rules-only-for-a-group-of-files). Have you tried this out yet? I would encourage you to play around with that configuration, and if you are able to find a solution, we could definitely add that helpful information to the README (only enabling eslint-plugin-jest for certain files using a shared config). Please let us know what your findings are! 😄

**Comment by alexkrolick** [comment]
That seems to work for those two rules, but you can't change `plugins` or `extends`, so if you have conflicts with the recommended config it might not work.

**Comment by SimenB** [comment]
You can specify `plugin`s, but not `extends`, in `overrides`.

You should either:

- manually compose the config in a js config file (you can do `require(eslint-plugin-jest).configs.recommended` and use that however you want)
- use multiple config files (with some `root: true` in them)
- wait for the resolution of https://github.com/eslint/rfcs/pull/9 or any alternative like it

I don't think it makes sense for us to figure out how to do our own filtering of rules on top of eslint's current mechanics for it. So I'll close, but feel free to continue the discussion. As mentioned above, a PR with instructions on how to do it is welcome if you think whatever solution you come up with should be shared :)

**Comment by alexkrolick** [comment]
Ok, documenting the override pattern sounds fine for now

**Comment by luan-orlandi-ifood** [comment]
I had the same issue, took me some time, but to solve this I did like @SimenB suggested: multiple .eslintrc files.

**Comment by tim-rohrer** [comment]
> I had the same issue, took me some time, but to solve this I did like @SimenB suggested: multiple .eslintrc files.

If you're still around, would you provide examples? I tried the suggestion but I appear to be doing something wrong as it doesn't work.

A snippet of my base `.eslintrc.json` in the project root:
```json
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:jest/recommended",
    "plugin:jest/style"
  ],
  "plugins": [
    "@typescript-eslint",
    "node",
    "jest"
  ],
  "overrides": [{
    "files": "./src/*"
  }],
```

And my complete `.eslintrc.json` file in the `cypress` folder (off my project root at the same level as `src`):

```json
{
  "extends": [
    "plugin:cypress/recommended"
  ],
  "plugins": [
    "cypress"
  ]
}
```

### Expected behavior:
Jest linting is provided inside `/src` and Cypress linting is provided inside `/cypress`. In part I expect this because the `overrides` should limit linting to `src`

### What is happening:
Jest linting rules continue to be applied to the Cypress test files.
 
### Current workaround:
Use `.eslintrc` to disable all rules inside `/cypress`. This is not ideal, but at least I can read the text in VSCode :-)

**Comment by G-Rath** [comment]
This is because you're still enabling `eslint-plugin-jest` for all files - `overrides` is effectively eslint configs + two extra properties for selecting files; you can actually have a config with just parser settings & `overrides` at the top level (which is one part of the new config stuff iirc, because frankly it's a better model).

So right now your override doesn't do anything, because you just have the file selector property. You need to remove `jest` from your `extends`, and put into an `extends` within your override block.

[Here's an example](https://github.com/G-Rath/audit-app/blob/main/.eslintrc.js#L13-L19)

**Comment by tim-rohrer** [comment]
>  `overrides` is effectively the standard eslint config "stuff" that you're use to having at the top level, + two extra properties for selecting files;

Woohoo! This was key to my understanding. I was confused about what all was supposed to go into `overrides`.

It isn't clear to me the Cypress rules are being applied as I don't have a single red squiggly line in the entire file, and I *doubt* that!

But, you got me pointed in the right direction, and for that I appreciate you. I'll head over to `eslint-plugin-cypress` to troubleshoot that.

**Comment by tim-rohrer** [comment]
Tonight I went to make some changes to a file and unit tests, but jest just seemed to stop running the prod files, and other weird behaviors. No errors. It took me a couple of hours, but I finally rolled back my `.eslintrc.json` changes, and jest started behaving properly again.

Before I open a new issue, will you look at my complete config and see if you see where I've done something stupid?

```json
{
  "root": true,
  "env": {
    "browser": false,
    "es2021": true,
    "jest": true
  },
  "settings": {
    "import/resolver": {
      "node": {
        "paths": ["src"],
        "extensions": [".js", ".ts"]
      }
    }
  },
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2021,
    "sourceType": "module"
  },
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended"
  ],
  "plugins": [
    "@typescript-eslint",
    "node"
  ],
  "overrides": [{
    "files": "./src/*",
    "plugins": ["jest"],
    "extends": [
      "plugin:jest/recommended ",
      "plugin:jest/style"
    ]
  }, {
    "files": "./test/*",
    "extends": [
      "plugin:cypress/recommended"
    ],
    "plugins": [
      "cypress"
    ]
  }],
  "rules": {
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": ["error"],
    "no-shadow": "off",
    "@typescript-eslint/no-shadow": ["error"],
    "import/extensions": 0,
    "lines-between-class-members": ["error", "always", {
      "exceptAfterSingleLine": true
    }],
    "quotes": ["error", "double"]
  }
}
```
Thank you.
